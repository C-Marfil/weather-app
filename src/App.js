function App() {
  return (
    <div className="props">
      <h1>Weather App</h1>
    </div>
  );
};

export default App;
