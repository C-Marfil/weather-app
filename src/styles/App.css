.weather-app {
    margin: 0 20px;
}