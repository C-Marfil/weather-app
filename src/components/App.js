import React, { useEffect, useState } from "react";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";
import SearchForm from "./SearchForm";
import "../styles/App.css";

const App = () => {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("");
  const [errorMessage, setErrorMessage] = useState("");
  const [selectedDescription, setDescription] = useState("welcome");
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  useEffect(() => {
    getForecast(setSelectedDate, setForecasts, setLocation);
  }, []);

  const handleForecastSelect = (date, description) => {
    setSelectedDate(date);
    setDescription(description);
  };

  const handleCitySearch = () => {
    getForecast(
      setSelectedDate,
      setForecasts,
      setLocation,
      searchText,
      setErrorMessage
    );
  };

  return (
    <div className="weather-app" id={selectedDescription}>
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </div>
  );
};

export default App;
