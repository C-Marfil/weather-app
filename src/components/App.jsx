import React from "react";
import PropTypes from "prop-types";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import "../styles/App.css";

function App({ location, forecasts }) {
  const { city, country } = location || {};
  return (
    <div className="weather-app">
      <LocationDetails city={city} country={country} />
      <ForecastSummaries forecasts={forecasts} />
    </div>
  );
}

App.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
    })
  ).isRequired,
  location: PropTypes.shape({
    city: PropTypes.string,
    country: PropTypes.string,
  }).isRequired,
};

export default App;
