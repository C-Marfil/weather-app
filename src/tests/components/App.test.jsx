import { render } from "@testing-library/react";
import React from "react";
import App from "../../components/App";

describe("App", () => {
  it("renders App as asserted", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
