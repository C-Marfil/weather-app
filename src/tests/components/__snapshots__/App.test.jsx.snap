// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App as asserted 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <div
      class="background-gif"
      id="welcome"
    />
    <p
      class="swipe-pomodoro"
    >
      Swipe Right for weather
    </p>
    <div
      class="pomodoro"
    >
      <div
        id="video"
        style="width: 100%; height: 100%;"
      />
    </div>
    <div
      class="nav"
    >
      <h1
        class="location-details"
      >
        , 
      </h1>
      <div
        class="search-form"
      >
        <input
          class="search-box"
          placeholder="Location"
          type="text"
          value=""
        />
        <button
          class="search-button"
          type="submit"
        >
          Set Location
        </button>
      </div>
      <div
        class="forecast-summaries"
      />
    </div>
  </div>
</DocumentFragment>
`;
